src/jit/lower.cpp

```diff
@@ -372,17 +372,16 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
         }
         break;
     case GT_CALL:
-        NYI("Call with TYP_LONG return value");
         break;
     case GT_RETURN:
-        assert(tree->gtOp.gtOp1->OperGet() == GT_LONG);
+        assert(tree->gtOp.gtOp1->OperGet() == GT_LONG || tree->gtOp.gtOp1->OperGet() == GT_CALL);
         break;
     case GT_STOREIND:
-        assert(tree->gtOp.gtOp2->OperGet() == GT_LONG);
+        assert(tree->gtOp.gtOp2->OperGet() == GT_LONG || tree->gtOp.gtOp2->OperGet() == GT_CALL);
         NYI("StoreInd of of TYP_LONG");
         break;
     case GT_STORE_LCL_FLD:
-        assert(tree->gtOp.gtOp1->OperGet() == GT_LONG);
+        assert(tree->gtOp.gtOp1->OperGet() == GT_LONG || tree->gtOp.gtOp1->OperGet() == GT_CALL);
```

Comment:
A suggestion to think about:

Since long decomposition happens during lowering, just before that we can force IR to be var=call for multi-reg call node cases.  We don't have to do that at all front-end phases.